Spawn st anchored in tabbed in same directory as current st. Inspired by
newterminal patch and dwm spawn command.
https://st.suckless.org/patches/newterm/

 config.def.h |  1 +
 st.c         | 16 ++++++++++++++++
 st.h         |  1 +
 3 files changed, 18 insertions(+)

diff --git a/config.def.h b/config.def.h
index 91ab8ca..d8bb42f 100644
--- a/config.def.h
+++ b/config.def.h
@@ -201,6 +201,7 @@ static Shortcut shortcuts[] = {
 	{ TERMMOD,              XK_Y,           selpaste,       {.i =  0} },
 	{ ShiftMask,            XK_Insert,      selpaste,       {.i =  0} },
 	{ TERMMOD,              XK_Num_Lock,    numlock,        {.i =  0} },
+	{ TERMMOD,              XK_Return,      tabbedst,       {.i =  0} },
 };

 /*
diff --git a/st.c b/st.c
index 134e724..4da2fdb 100644
--- a/st.c
+++ b/st.c
@@ -752,6 +752,22 @@ stty(char **args)
 		perror("Couldn't call stty");
 }

+void
+tabbedst(const Arg* a)
+{
+	if (fork() == 0) {
+		setsid();
+		/* Change directory */
+		char buf[32];
+		snprintf(buf, sizeof buf, "/proc/%d/cwd", pid);
+		chdir(buf);
+		/* Spawn tabbed and st */
+		const char* cmd[] = {"tabbed", "st", "-w", NULL};
+		execvp(((char **)cmd)[0], (char **)cmd);
+		die("execvp: %s\n", strerror(errno));
+	}
+}
+
 int
 ttynew(const char *line, char *cmd, const char *out, char **args)
 {
diff --git a/st.h b/st.h
index fd3b0d8..0b32a90 100644
--- a/st.h
+++ b/st.h
@@ -84,6 +84,7 @@ void draw(void);
 void printscreen(const Arg *);
 void printsel(const Arg *);
 void sendbreak(const Arg *);
+void tabbedst(const Arg *);
 void toggleprinter(const Arg *);

 int tattrset(int);
